superbench/benchmarks/micro_benchmarks/cuda_nccl_bw_performance.py

```diff
@@ -0,0 +1,139 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+"""Module of the NCCL performance benchmarks."""
+
+import os
+import re
+
+from superbench.common.utils import logger
+from superbench.benchmarks import BenchmarkRegistry, Platform, ReturnCode
+from superbench.benchmarks.micro_benchmarks import MicroBenchmarkWithInvoke
+
+
+class CudaNcclBwBenchmark(MicroBenchmarkWithInvoke):
+    """The NCCL bus bandwidth performance benchmark class."""
+    def __init__(self, name, parameters=''):
+        """Constructor.
+
+        Args:
+            name (str): benchmark name.
+            parameters (str): benchmark parameters.
+        """
+        super().__init__(name, parameters)
+
+        self._bin_name = 'all_reduce_perf'
+        self.__algorithms = {
+            'allreduce': 'all_reduce_perf',
+            'allgather': 'all_gather_perf',
+            'broadcast': 'broadcast_perf',
+            'reduce': 'reduce_perf',
+            'reducescatter': 'reduce_scatter_perf'
+        }
+
+    def add_parser_arguments(self):
+        """Add the specified arguments."""
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--algo',
+            type=str,
+            nargs='+',
+            default=list(self.__algorithms.keys()),
+            help='Nccl algorithms to benchmark. E.g. {}.'.format(' '.join(list(self.__algorithms.keys()))),
+        )
+        self._parser.add_argument(
+            '--gpu_count',
+            type=int,
+            default=8,
+            help='The count of GPUs used for benchmarking.',
+        )
+        self._parser.add_argument(
+            '--max_size',
+            type=str,
+            default='8192M',
+            help='Max size in bytes to run the nccl test. E.g. 8192M.',
+        )
+
+    def _preprocess(self):
+        """Preprocess/preparation operations before the benchmarking.
+
+        Return:
+            True if _preprocess() succeed.
+        """
+        if not super()._preprocess():
+            return False
+
+        # Format the arguments
+        if not isinstance(self._args.algo, list):
```

Comment:
In what situation, self._args.algo is not a list?

Reply:
I think if users use custom parameters and only want to test allreduce, he/she may write '--algo allreduce' such as what I defined in example file

Reply:
I think in this case, the self._args.algo will be also list, could you please double confirm?

Reply:
it's just string, not list

Reply:
If so, is this line a potential bug when specify one precision？
https://github.com/microsoft/superbenchmark/blob/702fb1eb3713b2a8c8d9070f7819bba7bab39be4/superbench/benchmarks/model_benchmarks/model_base.py#L297

Reply:
> If so, is this line a potential bug when specify one precision？
> https://github.com/microsoft/superbenchmark/blob/702fb1eb3713b2a8c8d9070f7819bba7bab39be4/superbench/benchmarks/model_benchmarks/model_base.py#L297

Oh, no, you're right, it's a list. the way I used to confirm is not right